src/baryvel.jl

```diff
@@ -0,0 +1,229 @@
+    sinf = sin(f)
+    cosf = cos(f)
+    dpsi = (1 - e^2) / (1 + e * cosf)
+    phid = 2 * e * 1.990969e-7 * ((1 + 1.5 * e^2) * cosf + e * (1.25 - 0.5 * sinf^2))
+    psid = (1.990969e-7 * e * sinf) / sqrt(1 - e^2)
+
+    # Perturbed heliocentric motion of the emb.
+    drd = (1 + pertr) * (psid + dpsi * pertrd)
+    drld = (1 + pertr) * dpsi * (1.990987e-7  + phid + pertld)
+    dtl = rem(dml + f - g + pertl, 2 * T(pi))
+    dsinls = sin(dtl)
+    dcosls = cos(dtl)
+    dxhd = drd * dcosls - drld * dsinls
+    dyhd = drd * dsinls + drld * dcosls
+
+    # Influence of eccentricity, evection and variation on the geocentric
+    a = rem.(dcargm[:, 1] + dt * dcargm[:, 2], 2 * T(pi))
+    sina = sin.(a)
+    cosa = cos.(a)
+    pertl_m = zero(T)
+    pertld_m = zero(T)
+    pertp_m = zero(T)
+    pertpd_m = zero(T)
+    for i = 1:3
+        pertl_m += ccampm[i, 1] * sina[i]
+        pertld_m += ccampm[i, 2] * cosa[i]
+        pertp_m += ccampm[i, 3] * cosa[i]
+        pertpd_m += - ccampm[i, 4] * sina[i]
+    end
+
+    # Heliocentric motion of the earth.
+    tl = forbel[2] + pertl_m
+    sinlm = sin(tl)
+    coslm = cos(tl)
+    sigma = 3.122140e-5 / (1 + pertp_m)
+    a = sigma * (2.661699e-6 + pertld_m)
+    b = sigma * pertpd_m
+    dxhd += a * sinlm + b * coslm
+    dyhd -= a * coslm + b * sinlm
+    dzhd= -sigma * cos(forbel[3]) * 2.399485e-7
+
+    # Barycentric motion of the earth.
+    dxbd = dxhd * 0.99999696
+    dybd = dyhd * 0.99999696
+    dzbd = dzhd * 0.99999696
+    for i = 1:4
+        plon = forbel[i+3]
+        pomg = sorbel[i+1]
+        pecc = sorbel[i+9]
+        tl = rem(plon + 2 * pecc * sin(plon-pomg), 2 * T(pi))
+        dxbd += ccpamv[i] * (sin(tl) + pecc * sin(pomg))
+        dybd -= ccpamv[i] * (cos(tl) + pecc * cos(pomg))
+        dzbd -= ccpamv[i] * sorbel[i+13] * cos(plon - sorbel[i+5])
+    end
+
+    # Transition to mean equator of date.
+    dcosep = cos(deps)
+    dsinep = sin(deps)
+    dyahd = dcosep * dyhd - dsinep * dzhd
+    dzahd = dsinep * dyhd + dcosep * dzhd
+    dyabd = dcosep * dybd - dsinep * dzbd
+    dzabd = dsinep * dybd + dcosep * dzbd
+
+    if deq == 0
+        dvelh = AU/1000 * ([dxhd, dyahd, dzahd])
+        dvelb = AU/1000 * ([dxbd, dyabd, dzabd])
+    else
+        deqdat = ((dje - 2415020.313) / 365.24219572) + 1900
+        prema = premat(deqdat, deq, FK4 = true)
+        dvelh = AU/1000 * (prema * [dxhd, dyahd, dzahd])
+        dvelb = AU/1000 * (prema * [dxbd, dyabd, dzabd])
+    end
+    return dvelh, dvelb
+end
+
+"""
+    baryvel(dje, deq) -> dvelh, dvelb
+
+### Purpose ###
+
+Calculates heliocentric and barycentric velocity components of Earth.
+
+### Explanation ###
+
+Baryvel takes into account the Earth-Moon motion, and is useful for radial velocity
+work to an accuracy of ~1 m/s.
+
+### Arguments ###
+
+* `dje`: julian ephemeris date
+* `deq`: epoch of mean equinox of `dvelh` and `dvelb`. If deq=0, then deq is
```

Comment:
Rather than doing different things depending on the value of `deq`, you can do something more Julian by dispatching the function based on the number of arguments.

Very crude example:
```julia
baryvel(dje::Real) = _baryvel(float(dje))
baryvel(dje::Real, deq::Real) = _baryvel(promote(float(dje), float(deq))...)

function _baryvel(dje::T)
    v1, v2 = foo(dje)
    dvelh = v1 * AU / 1000
    dvelb = v2 * AU / 1000
end

function _baryvel(dje::T, deq::T)
    v1, v2 = foo(dje)
    deqdat = ((dje - 2415020.313) / 365.24219572) + 1900
    prema = premat(deqdat, deq, FK4 = true)
    dvelh = (prema * v1) * AU / 1000
    dvelb = (prema * v2) * AU / 1000
end
```
where `foo(dje)` (find a better name ;-) is the function returning the 2-tuple of vectors
```julia
[dxhd, dyahd, dzahd], [dxbd, dyabd, dzabd]
```